src/App.tsx

```diff
@@ -1,41 +1,116 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './App.scss';
 import './styles/general.scss';
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
+import { getPostDetails, getUserPosts, getUsers } from './api/posts';
+import { addPostComent, getPostComments, removeComments } from './api/comments';
+import { Post } from './types';
 
-const App: React.FC = () => (
-  <div className="App">
-    <header className="App__header">
-      <label>
-        Select a user: &nbsp;
-
-        <select className="App__user-selector">
-          <option value="0">All users</option>
-          <option value="1">Leanne Graham</option>
-          <option value="2">Ervin Howell</option>
-          <option value="3">Clementine Bauch</option>
-          <option value="4">Patricia Lebsack</option>
-          <option value="5">Chelsey Dietrich</option>
-          <option value="6">Mrs. Dennis Schulist</option>
-          <option value="7">Kurtis Weissnat</option>
-          <option value="8">Nicholas Runolfsdottir V</option>
-          <option value="9">Glenna Reichert</option>
-          <option value="10">Leanne Graham</option>
-        </select>
-      </label>
-    </header>
-
-    <main className="App__main">
-      <div className="App__sidebar">
-        <PostsList />
-      </div>
-
-      <div className="App__content">
-        <PostDetails />
-      </div>
-    </main>
-  </div>
-);
+interface User {
+  name: string;
+  id: number;
+}
+
+const App: React.FC = () => {
+  const [posts, setPosts] = useState([]);
+  const [users, setUsers] = useState([]);
+  const [selectUser, setSelectUser] = useState(0);
+  const [postId, setPostId] = useState(0);
+  const [postDetails, setPostDetails] = useState<Post | null>(null);
+  const [postComments, setPostComments] = useState([]);
+
+  const changePostId = (id: number) => {
+    setPostId(id);
+  };
+
+  useEffect(() => {
+    getUsers().then(setUsers);
+  }, []);
+
+  useEffect(() => {
+    getPostComments(postId).then(result => {
+      setPostComments(result);
+    });
+
+    getPostDetails(postId).then(result => {
+      setPostDetails(result);
+    }).catch(() => {
+      setPostDetails(null);
+    });
```

Comment:
you should fetch post comments and details in PostDetails component, and only when the component is mounting or the postId is changed